lib/dateHelpers.ts

```diff
@@ -12,7 +15,34 @@ const timestampFromDatetimeLocal = (datetimeLocal: string): Long => {
     Number(hours),
     Number(minutes),
   );
-  return Long.fromNumber(dateObj.getTime()).divide(1000);
+
+  const timestampMillis = Long.fromNumber(dateObj.getTime());
+
+  switch (units) {
+    case "s":
+      return timestampMillis.divide(1000); // seconds
+    case "ns":
+      return timestampMillis.divide(1000_000); // nanoseconds
```

Comment:
Here you need to multiply to make the number bigger. E.g.

```
> let timestampMillis = Date.now()
undefined
> timestampMillis
1686840184039
> timestampMillis * 1_000_000
1686840184039000000
```

The `return timestampMillis.divide(1000); // seconds` line is correct